src/settings/rule_table.rs

```diff
@@ -0,0 +1,55 @@
+use std::collections::hash_map;
+
+use rustc_hash::FxHashMap;
+
+use super::hashable::HashableHashMap;
+use crate::registry::RuleCode;
+
+/// A table to keep track of which rules are enabled
+/// and whether or not they should be autofixed.
+#[derive(Debug, Hash)]
+pub struct RuleTable {
+    /// Maps rule codes to a boolean indicating if the rule should be autofixed.
+    enabled: HashableHashMap<RuleCode, bool>,
+}
+
+impl RuleTable {
+    /// Creates a new empty rule table.
+    pub fn empty() -> Self {
+        Self {
+            enabled: HashableHashMap::default(),
+        }
+    }
+
+    /// Returns whether the given rule should be checked.
+    pub fn enabled(&self, code: &RuleCode) -> bool {
+        self.enabled.contains_key(code)
```

Comment:
Do you still plan on changing this to use macro-based lookups or no longer viable?

Reply:
No I don't think macro-based lookups make sense (especially considering that they don't give any substantial performance gain). I think we should strive to decouple the ruff core from our linter implementations and requiring one boolean field in a struct for every rule is very much the opposite of decoupling.